mmv1/products/alloydb/Cluster.yaml

```diff
@@ -179,12 +179,37 @@ properties:
             output: true
   - !ruby/object:Api::Type::String
     name: 'network'
-    required: true
+    exactly_one_of:
+      - network
+      - network_config.0.network
+    default_from_api: true
+    deprecation_message: >-
+      `network` is deprecated and will be removed in a future major release. Instead, use `network_config` to define the network configuration.
     description: |
       The relative resource name of the VPC network on which the instance can be accessed. It is specified in the following form:
 
       "projects/{projectNumber}/global/networks/{network_id}".
     diff_suppress_func: 'tpgresource.ProjectNumberDiffSuppress'
+  - !ruby/object:Api::Type::NestedObject
+    name: 'networkConfig'
```

Comment:
nit: Can we use the REST documentation wording as it is here and for all the sub-fields as well? [reference](https://cloud.google.com/alloydb/docs/reference/rest/v1beta/projects.locations.clusters#networkconfig).